### PR TITLE
FEAT: use halo2-ecc's ecdsa implementation [Ready for review]

### PR DESCRIPTION
**Pull request by zhenfeizhang**

### CODE REVIEW

File: zkevm-circuits/src/tx_circuit/sign_verify.rs [hunk 1]
```diff
@@ -115,8 +132,9 @@ const BIT_LEN_LAST_LIMB: usize = 256 - (NUMBER_OF_LIMBS - 1) * BIT_LEN_LIMB;
 
 /// SignVerify Configuration
 #[derive(Debug, Clone)]
-pub(crate) struct SignVerifyConfig {
+pub(crate) struct SignVerifyConfig<F: Field> {
     // ECDSA
+    ecdsa_config: FpChip<F>,
     main_gate_config: MainGateConfig,
     range_config: RangeConfig,
```
Comment (kunxian-xia) [inline zkevm-circuits/src/tx_circuit/sign_verify.rs:139]: We need to remove the configs used by halo2wrong.

File: zkevm-circuits/src/tx_circuit/sign_verify.rs [hunk 2]
```diff
@@ -128,13 +146,40 @@ pub(crate) struct SignVerifyConfig {
...
+            91,
+            3,
+            modulus::<Fp>(),
+            "ecdsa chip".to_string(),
+        );
+
+        // halo2wrong's ECDSA config
         let (rns_base, rns_scalar) =
             GeneralEccChip::<Secp256k1Affine, F, NUMBER_OF_LIMBS, BIT_LEN_LIMB>::rns();
         let main_gate_config = MainGate::<F>::configure(meta);
```
Comment (kunxian-xia) [inline zkevm-circuits/src/tx_circuit/sign_verify.rs:185]: Need to remove these too.

File: zkevm-circuits/src/tx_circuit/sign_verify.rs (hunk 1 above)
Comment (zhenfeizhang) [reply zkevm-circuits/src/tx_circuit/sign_verify.rs]: removed range_config etc. still need to use main_gate_config for keccak lookup

File: zkevm-circuits/src/tx_circuit/sign_verify.rs [hunk 3]
```diff
@@ -306,71 +236,45 @@ impl<F: Field> Term<F> {
...
+pub(crate) struct AssignedECDSA<F: Field, FC: FieldChip<F>> {
+    pk: EccPoint<F, FC::FieldPoint>,
+    msg_hash: OverflowInteger<F>,
+    sig_is_valid: halo2_base::AssignedValue<F>,
 }
 
 #[derive(Debug)]
 pub(crate) struct AssignedSignatureVerify<F: Field> {
-    pub(crate) address: AssignedValue<F>,
+    pub(crate) address: halo2_base::AssignedValue<F>,
```
Comment (kunxian-xia) [inline zkevm-circuits/src/tx_circuit/sign_verify.rs:247]: I think we need to include `sig_is_valid` in this struct s.t. tx circuit can assert it's always true.

**Comment by silathdiir** [comment]
Suppose to move to https://github.com/scroll-tech/zkevm-circuits/pull/282, since private crate access `halo2-ecc` is only configured on `scroll-tech/zkevm-circuits` (not `zhengfeizhang/zkevm-circuits`).

**Pull request closed without merging**